### PR TITLE
[MERGE] *: new OWL reporting views

### PR DESCRIPTION
**Pull request by Polymorphe57**
In this PR, we introduce the new OWL view infrastructure and convert
the graph and pivot views to OWL. More details are given in each commit.

Co-authored-by: Aaron Bohy <aab@odoo.com>
Co-authored-by: Bruno Boi <boi@odoo.com>
Co-authored-by: Géry Debongnie <ged@odoo.com>
Co-authored-by: Samuel Degueldre <sad@odoo.com>
Co-authored-by: Mathieu Duckerts-Antoine <dam@odoo.com>
Co-authored-by: Simon Genin (ges) <ges@odoo.com>
Co-authored-by: Francois (fge) <fge@odoo.com>
Co-authored-by: Michael Mattiello (mcm) <mcm@odoo.com>
Co-authored-by: Julien Mougenot <jum@odoo.com>
Co-authored-by: Lucas Perais (lpe) <lpe@odoo.com>
Co-authored-by: Jorge Pinna Puissant <jpp@odoo.com>
Co-authored-by: stefanorigano (SRI) <sri@odoo.com>

### CODE REVIEW

File: odoo/models.py [hunk 1]
```diff
@@ -3079,6 +3079,7 @@ def fields_get(self, allfields=None, attributes=None):
                 continue
 
             description = field.get_description(self.env)
+            description['name'] = fname
```
Comment (rco-odoo) [inline odoo/models.py:3082]: :+1:

File: addons/web/static/src/views/graph/graph_renderer.xml [hunk 2]
```diff
@@ -24,7 +24,7 @@
 
     <t t-name="web.GraphRenderer" owl="1">
         <div class="o_graph_renderer">
-            <label t-if="props.metaData.title" t-esc="props.metaData.title"/>
+            <!-- <label t-if="props.metaData.title" t-esc="props.metaData.title"/> -->
```
Comment (Polymorphe57) [inline addons/web/static/src/views/graph/graph_renderer.xml:27]: problem with style --> we should review the use of string/title attribute on arch through time (from 12.0 to now)

File: addons/board/static/tests/dashboard_tests.js [hunk 3]
```diff
@@ -1053,7 +1062,8 @@ QUnit.test('click on a cell of pivot view inside dashboard', async function (ass
     form.destroy();
 });
 
-QUnit.test(
+// TODO: The button "Add to my dashboard" is not yet developped on the new control panel search view
+QUnit.skip(
```
Comment (Polymorphe57) [inline addons/board/static/tests/dashboard_tests.js:1066]: unskip now

File: addons/web/static/src/core/utils/objects.js [hunk 4]
```diff
@@ -10,3 +10,7 @@ export function shallowEqual(obj1, obj2) {
         obj1Keys.every((key) => obj1[key] === obj2[key])
     );
 }
+
+export function deepCopy(obj) {
```
Comment (aab-odoo) [inline addons/web/static/src/core/utils/objects.js:14]: This must be documented, as the function isn't complete. It is fast, which is good, but it doesn't work on everything, so it must be explicitly written in the docstring

File: addons/web/static/src/views/view.xml [hunk 5]
```diff
@@ -5,4 +5,23 @@
     <WithSearch t-props="withSearchProps" class="o_action o_view_controller"/>
   </t>
 
+  <t t-name="web.ReportViewMeasures" owl="1">
```
Comment (aab-odoo) [inline addons/web/static/src/views/view.xml:8]: I would expect this commit to remove several occurrences of that template

**Comment by Polymorphe57** [comment]
3 skipped tests:
 - 2 in spreadsheets --> might be linked to change in mockServer --> date format for weeks?
 - 1 in appointment: test of a calendar view

**Comment by kebeclibre** [comment]
@odoo/rd-security Can you approve the few security risks ?
The first one is completely secure since renderToString escapes the text within the template
The second and third are not completely secure, but it was already done that way before those commits, and we sure plan on thinking of another way after the merge

thanks guys

**Comment by kebeclibre** [comment]
robodoo rebase-ff

**Comment by odony** [comment]
@robodoo override=ci/security
Cfr explanations by @kebeclibre, but pending further double-check post-freeze.

Note: using `Markup` + `t-out` should replace `t-raw`, not sure what are the plans for supporting this in owl? It would be nice to be consistent here for v15, considering that non-own JS and Python code have been converted.  /cc @kebeclibre @aab-odoo @xmo-odoo

**Comment by Polymorphe57** [comment]
robodoo r+

**Comment by odony** [comment]
@robodoo delegate=Polymorphe57

**Comment by Polymorphe57** [comment]
robodoo r+

**Comment by Polymorphe57** [comment]
robodoo retry

**Comment by xmo-odoo** [comment]
> Note: using Markup + t-out should replace t-raw, not sure what are the plans for supporting this in owl? It would be nice to be consistent here for v15, considering that non-own JS and Python code have been converted. /cc @kebeclibre @aab-odoo @xmo-odoo

After the merge of the qwebjs bits @aab-odoo told me they were taking charge of the owl part. There's a task in the JS project (2610127) but don't have any news.

**Comment by aab-odoo** [comment]
robodoo rebase-merge r+

**Comment by aab-odoo** [comment]
robodoo r-

**Comment by aab-odoo** [comment]
robodoo r+

File: addons/board/static/tests/dashboard_tests.js (hunk 3 above)
Comment (mart-e) [reply addons/board/static/tests/dashboard_tests.js:1066]: still skipped ?

**Pull request closed without merging**